### ISSUE TITLE
IOT Box version 20_10 to 21_10 Star POP10 (STR-001) printing error

### ISSUE THREAD

**Issue by tubndgit** [odoo/odoo#82872]
Hello All,
I'm using IOT Box version from 20_10 to 21_10 to test POS receipt printing with Star POP10 but it runs to character errors (Attached images)
Can you all help me on this issue? Thank you very much!
![IMAG0293](https://user-images.githubusercontent.com/5570339/149744828-9f153d47-f996-48b4-a3bc-5ce8010eec43.jpg)
![IMAG0294](https://user-images.githubusercontent.com/5570339/149745048-754c0875-2f3f-4c6d-a818-f84a97d5210c.jpg)

**Comment by qle-odoo** [comment]
Hello,
these characters appear on all tickets?
Or only on the status ticket?

Can you test with another printer (other than STAR?)

**Comment by tubndgit** [comment]
Hello,
It only appears on receipt, not on status ticket

**Comment by qle-odoo** [comment]
Maybe like Star TSP650II ?
The Epson printers work out of the box. 
The Star one requires that you set some DIP switches on the bottom to put it in ESC/POS mode. 
By default it's in Star mode which is their own protocol that we don't support.
unscrew the small panel at the bottom
set the dipswitches as follows:
. . . .     . .     .   .
          .     . .   .
1 2 3 4   1 2 3 4 5 6 7 8

https://www.oki.com/printing/download/46247601EE1_RT322_UG_EN_14571.pdf?id=46247601EE

**Comment by tubndgit** [comment]
Hi @qle-odoo 
Thank you very much for your message, but i found this document that doesn't support for Star POP 10
https://www.starmicronics.com/help-center/knowledge-base/how-to-switch-emulations-on-star-desktop-printers/

Also I have changed a bit on this file and it can cut paper but the charracters still error

`hw_drivers/iot_handlers/drivers/PrinterDriver.py`

Change:
`self.receipt_protocol = 'star' if 'STR_T' in device['device-id'] else 'escpos'`
To:
`self.receipt_protocol = 'star' if 'STR' in device['device-id'] else 'escpos'`

**Comment by tubndgit** [comment]
Hi @qle-odoo 
I found this [document ](https://www.starmicronics.com/support/Mannualfolder/StarPRNT_cm_en.pdf)for, it is a command specification for mPOP such as model POP10, can you help me. Thank you very much!

**Comment by qle-odoo** [comment]
Hello,
With the IoT we use cups to send escpos to printer
Have see this document from STAR ?
http://www.starmicronics.com/support/mannualfolder/common_tsp_linux_en.pdf
By default driver is already installed but you can try a print test page or change driver to the STAR driver

**Comment by tubndgit** [comment]
Hi @qle-odoo 
The version of cups on iotbox is CUPS 2.2.10, does it support STAR mPOP (POP10)

**Comment by tubndgit** [comment]
Hi @qle-odoo 
I can not see driver for mpop POP 10 so I have uploaded driver file for it (.ppd) 
The receipt characters is still error
I used iotbox version 21.10 and the cups is 2.2.10
![image](https://user-images.githubusercontent.com/5570339/150059583-d503ed8c-16a6-40dc-a223-34b1d82c5f87.png)

Any idea on it?
Thank you

**Comment by tubndgit** [comment]
Hi @qle-odoo 
Are you working on this issue?
How can I update cups and star driver on iotbox?
Thank you!

**Comment by qle-odoo** [comment]
You test a print page with cups ?

**Comment by tubndgit** [comment]
No, I want to upgrade cups to latest version and upgrade Star driver http://www.starmicronics.com/support/default.aspx?printerCode=CUPS_for_Linux

Currently they are out of date

**Comment by qle-odoo** [comment]
normaly the cups version is up to date on buster version.
https://packages.debian.org/buster/cups

**Comment by tubndgit** [comment]
Yes, how about Star driver?
![150060087-9121023b-1373-4ee3-b126-bff2c5532ce2 1](https://user-images.githubusercontent.com/5570339/150976157-71b9f7bb-fe44-454d-9d72-bd3a94a6c057.png)

**Comment by qle-odoo** [comment]
Maybe you can put manually ?
Put the box in writing mode
sudo mount -o remount,rw /
sudo mount -o remount,rw /root_bypass_ramdisks/

**Comment by tubndgit** [comment]
Hello @qle-odoo 
Thank you for your reply. I have tried this command and upgraded Star driver but the receipt characters still error
You can see the attached image when I try to print test page on CUPS

![IMAG0298](https://user-images.githubusercontent.com/5570339/151480345-d10c85bd-4395-408d-b3b6-71b0f9a40f03.jpg)

**Comment by qle-odoo** [comment]
Ok the print page is a PDF.
So if you are able to print a PDF you are not able to print a escpos document (like a receipt)
I think it is a driver issue.

**Comment by tubndgit** [comment]
Hello @qle-odoo 
Thank you for your support!
These are any solutions to make this printer work with PoS receipt printing?

**Comment by nmathon** [comment]
Hi @tubndgit,
Did you find a solution to this? I am also using mPop and wonder if it can be connected to thge IOT box.
Thanks

**Comment by nuttydelights** [comment]
Did you found any solution to this ?  I am also using mPop using IOT box in odoo

**Comment by qle-odoo** [comment]
For resume:
You can try with lastest iot box 25_01 [https://nightly.odoo.com/master/iotbox/](https://nightly.odoo.com/master/iotbox/)
In this new image cups is in the latest version with all printer drivers available for raspberry installed.
If it still doesn't work you can try to change drivers in cups.
Or directly download and install an unofficial Debian driver from Star maybe.

Other solution use IoT Windows (it's free) setup is same.
If you are able to print with your Windows computer so your IoT windows too.
https://www.odoo.com/documentation/18.0/fr/applications/general/iot/windows_iot.html

We can't guarantee the operation of Star mPop printers and I refer you to the printers recommended by Odoo.
